Plugins/TGeo/src/TGeoParser.cpp

```diff
@@ -73,18 +73,26 @@ void Acts::TGeoParser::select(Acts::TGeoParser::State& state,
                 dynamic_cast<TGeoBBox*>(state.node->GetVolume()->GetShape());
             shape != nullptr) {
           // @TODO this should be replace by a proper TGeo to Acts::VolumeBounds
-          // and vertices converision which would make a more appropriate
-          // parsomg
-          double dx = options.unit * shape->GetDX();
-          double dy = options.unit * shape->GetDY();
-          double dz = options.unit * shape->GetDZ();
+          // and vertices conversion which would make a more appropriate parsing
+          const double dx = options.unit * shape->GetDX();
+          const double dy = options.unit * shape->GetDY();
+          const double dz = options.unit * shape->GetDZ();
           for (auto x : std::vector<double>{-dx, dx}) {
+            if (!accept) {
+              break;
+            }
             for (auto y : std::vector<double>{-dy, dy}) {
```

Comment:
since the loops seem independent something like a `product` iteration like in python might be nice. then the control flow is easier to handle with a single break. otherwise I agree with paul that the additional looping is not in a critical path so the old version might be preferable